### PR TITLE
Refactor/error handling

### LINKED ISSUE #13: Error Handling
**Describe the User Story associated with this feature**
We need to add error handling for the following:
   - A user should see a 404 error when the url path doesn't exist
   - A user should see a local error message when something doesn't load correctly
   - If a user visits an empty favorites page, show a message asking them to add some favorites.
   - Display an error when a fetch fails
   - A user should not be able to duplicate their favorites (maybe disable the button? If ambitious, we could toggle to button to to 'remove from favorites' instead)
 
**Describe the acceptance criteria**
This issue can be closed when the above errors are addressed.

### PR DESCRIPTION
**Pull request by whiteheadol**
# Description

Error handling PR. Accounts for failed fetches, user errors, and removes the ability to favorite a piece more than once. 

## Issues

Closes #13

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [x] Cosmetic changes (changes to layout or visuals in CSS file)

## How Has This Been Tested?

- [x] console.log()
- [x] dev tools
- [x] functionality check through web browser
- [ ] tested by viewing the markdown file preview
- [ ] tested with testing framework (Cypress)

## Checklist:

- [x] My code follows the style guidelines of this project (semicolons, indentation, naming conventions, etc..)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have deleted all comments/console.log() after functionality and understanding is final
- [x] I have added any issues closed by this PR in the Issues section.
- [x] I have added others as reviewers

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,31 +1,36 @@
...
+        }} />
+        <Route path='/favorites' render={() => {
+          return (
+            <div>
+              <FavoriteNav />
+              <FavoritesView />
+            </div>
+          )
+        }} />
+        <Route path="*" component={Error}/>
```
Comment (Jlavarine) [inline src/App.js:32]: Good thinking implementing a 404 page

**Review by Jlavarine — APPROVED** [review]
Lots of great error handling

**Review by nbalmoori — APPROVED** [review]
Great error handling!

**Review by DanielN88 — APPROVED** [review]
Everything looks good, the button disable is working as intended. Ready for a merge.

**Comment by nbalmoori** [comment]
Looks great!

**Pull request merged**